group/curve25519/curve.go

```diff
@@ -391,3 +388,16 @@ func (c *curve) data(x, y *mod.Int) ([]byte, error) {
 	}
 	return b[1 : 1+dl], nil
 }
+
+// reverse copies src into dst in byte-reversed order and returns dst,
+// such that src[0] goes into dst[len-1] and vice versa.
+// dst and src may be the same slice but otherwise must not overlap.
+func reverse(dst, src []byte) []byte {
+	l := len(dst)
+	for i, j := 0, l-1; i < (l+1)/2; {
```

Comment:
code:
```
for i, j := 0, l-1; i < (l+1)/2; i++, j--
```

Reply:
That's not vaild Go. Only one expression is allowed after the second semi-colon. (surprised me too!)

Reply:
Now I see how they did it at https://github.com/golang/go/wiki/SliceTricks :
```
for i, j := 0, l-1; i < (l+1)/2; i, j = i+1, j-1
```